src/spec_concordius.cpp

```diff
@@ -0,0 +1,85 @@
+/************************************************************************/
+/*  Xania (M)ulti(U)ser(D)ungeon server source code                     */
+/*  (C) 2020 Xania Development Team                                     */
+/*  See the header to file: merc.h for original code copyrights         */
+/************************************************************************/
+
+#include "CHAR_DATA.hpp"
+#include "DescriptorFilter.hpp"
+#include "DescriptorList.hpp"
+#include "comm.hpp"
+#include "merc.h"
+#include "string_utils.hpp"
+#include <fmt/format.h>
+#include <string_view>
+#include <vector>
+
+using namespace fmt::literals;
+namespace {
+
+static inline constexpr auto SOCIAL_DICE = 30u;
+static inline constexpr auto NUM_SOCIAL_ACTIONS = 7u;
+
+inline static const std::vector<std::string> emotes{
+    "$n polishes $s golden claw.",
+    "$n looks skyward and closes $s eyes in thought.",
+    "$n raises $s hand and begins to whistle birdsong.",
+    "$n says 'I used to be an adventurer like you. But then Aquila talked some sense into me.'",
+    "$n sits down and begins to sharpen the edge of his shield against his heel.",
+    "$n says 'Beware the denizens of Yggdrasil to the west!'",
+    "$n tightens $s toga.",
+    "$n says 'How I long to visit Ultima once again...",
+    "$n says 'I'm told it's never sunny in Underdark. I don't like the sound of that.'",
+    "$n says 'Did you know there are over sixty different species of eagle?!'"};
+
+bool concordius_stands_guard(CHAR_DATA *ch) {
+
+    uint random = number_range(0, 100);
+    if (random > 90) {
+        act(emotes[random % emotes.size()], ch, nullptr, nullptr, To::Room);
+        return true;
+    }
+    bool found_someone = false;
+    for (auto &victim : descriptors().playing() | DescriptorFilter::same_room(*ch) | DescriptorFilter::to_character()) {
+        uint random = number_range(0, SOCIAL_DICE);
+        if (random < SOCIAL_DICE - NUM_SOCIAL_ACTIONS)
+            continue;
+        found_someone = true;
+        std::string emote{};
+        switch (random % NUM_SOCIAL_ACTIONS) {
+        default:
+        case 0: emote = "bow {}"_format(victim.name); break;
```

Comment:
overkill perhaps but maybe to avoid the spooky-action-at-a-distance of `NUM_SOCIAL_ACTIONS` and this list:

```cpp

struct ConcordiusPersonalEmote {
  std::string_view on_good;
  std::string_view otherwise;
};
std::array personal_emotes = {
ConcordiusPersonalEmote{"eye", "eye"},
ConcordiusPersonalEmote{"salute", "salute"},
ConcordiusPersonalEmote{"punch", "judge"},
//..etc
};

//...
auto &pers_emote = personal_emotes[random % personal_emotes.size()];
interpret(ch, "{} {}"_format(victim.is_good() ? pers_emote.on_good : pers_emote.otherwise, victim.name).c_str());
```

(untested! obviously ... and ironically the std:;array thing is annoying here, a vector <might> be better after all!) just a thought anyway

Reply:
Yeah funnily enough my first thought was to use an 'enum' but that was Java thinking. But I don't really like switch statements like this, I will try to improve it.

Reply:
One of the interesting side effects of how things are set up is that the more players there are in the room, the more social he will seem.